ext/opcache/jit/zend_jit_vm_helpers.c

```diff
@@ -613,20 +613,38 @@ static int zend_jit_trace_subtrace(zend_jit_trace_rec *trace_buffer, int start,
  *  Trace Linking Rules
  *  ===================
  *
- *                                          flags
- *          +----------+----------+----------++----------+----------+----------+
- *          |                                ||              JIT               |
- *          +----------+----------+----------++----------+----------+----------+
- *   start  |   LOOP   |  ENTER   |  RETURN  ||   LOOP   |  ENTER   |  RETURN  |
- * +========+==========+==========+==========++==========+==========+==========+
- * | LOOP   |   loop   |          | loop-ret || COMPILED |   LINK   |   LINK   |
- * +--------+----------+----------+----------++----------+----------+----------+
- * | ENTER  |INNER_LOOP| rec-call |  return  ||   LINK   |   LINK   |   LINK   |
- * +--------+----------+----------+----------++----------+----------+----------+
- * | RETURN |INNER_LOOP|          |  rec-ret ||   LINK   |          |   LINK   |
- * +--------+----------+----------+----------++----------+----------+----------+
- * | SIDE   |  unroll  |          | side-ret ||   LINK   |   LINK   |   LINK   |
- * +--------+----------+----------+----------++----------+----------+----------+
+ *  We consider the start flags of the current trace, as well as the trace flags
+ *  of the opline to be recorded.
+ *
+ *                         opline trace flags
```

Comment:
Isn't this more of an event, rather than flags? E.g. we check `execute_data->prev_execute_data == prev_execute_data` to see whether we have entered a function.